actor/src/main/scala-2.13/org/apache/pekko/util/ByteString.scala

```diff
@@ -702,6 +728,33 @@ object ByteString {
       }
     }
 
+    override def indexOf(elem: Byte, from: Int): Int = {
+      if (from >= length) -1
+      else {
+        val byteStringsSize = bytestrings.size
+
+        @tailrec
+        def find(bsIdx: Int, relativeIndex: Int, bytesPassed: Int): Int = {
+          if (bsIdx >= byteStringsSize) -1
+          else {
+            val bs = bytestrings(bsIdx)
+
+            if (bs.length <= relativeIndex) {
+              find(bsIdx + 1, relativeIndex - bs.length, bytesPassed + bs.length)
+            } else {
+              val subIndexOf = bs.indexOf(elem, relativeIndex)
+              if (subIndexOf < 0) {
+                val nextString = bsIdx + 1
+                find(nextString, relativeIndex - bs.length, bytesPassed + bs.length)
```

Comment:
or we just inline it to the arguments.

Reply:
Went with this solution in e923cba. This is also what happens in the branch above, so the code looks more symmetrical now. 👍